Added Field type reference (#7910)
magento/devdocs: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md

{:.bs-callout-warning}
The nodes `more_url`, `demo_url`, `requires` and `options` are defined by a different core payment model and are only used once. These nodes are not reusable.

### Field input type reference

> Each field has the default type of ´text´ to display it's frontend input type. The following input fields types can be found and used in the core at the time of this writing:

Reviewer: ```suggestion
Each field has the default type of ´text´ to display its frontend input type. The following input field types can be found and used in the core at this time :
```

| Field Type    | Description                                                                                                       |
|---------------|-------------------------------------------------------------------------------------------------------------------|
| ´text´        | Default. Standard text field.                                                                                     |

> | ´select´      | A common HTML select. Only one item can be selected.                                                              |

Reviewer: It should also be mentioned that this covers a YES/NO and it requires a source_model to manage the items on the list

EG - Normal dropdown, may need custom `source_model` - also used for Yes/No. See `Magento\Search\Model\Adminhtml\System\Config\Source\Engine` for an example

> | ´select´      | A common html select. Only one item can be selected.                                                              |

Reviewer: ```suggestion
| ´select´      | A common HTML select. Only one item can be selected.                                                              |
```

> | ´multiselect´ | A common HTML select. Multiple items can be selected at once.                                                     |

Reviewer: Also requires a source_model

> | ´multiselect´ | A common html select. Multiple items can be selected at once.                                                     |

Reviewer: ```suggestion
| ´multiselect´ | A common HTML select. Multiple items can be selected at once.                                                     |
```

> | ´obscure´     | Commonly used for passwords. Shows asterisks as values.                                                           |

Reviewer: Also mention that the value is encrypted and forcibly changing type using "Inspect Element" in browser will not show the value

> | ´time´        | A string of the current time. Cannot be configured.                                                               |

Reviewer: Incorrect definition : Allows a time to be set. This is done by using 3 dropdowns - Hour, minute and second.

> | ´textarea´    | A common HTML textarea.                                                                                           |

Reviewer: Would suggest that this be moved to under 'text'

> | ´textarea´    | A common html textarea.                                                                                           |

Reviewer: ```suggestion
| ´textarea´    | A common HTML textarea.                                                                                           |
```

> | ´button´      | A common html button. It is necessary to declare a custom ´frontend_model´ in order to use the button field type. | 

Reviewer: ```suggestion
| ´button´      | A common HTML button. It is necessary to declare a custom ´frontend_model´ in order to use the button field type. | 
```

Reviewer: Would also recommend mentioning : Requires custom front-end model to define the button text and the action. See `Magento\ScheduledImportExport\Block\Adminhtml\System\Config\Clean` for an example

> | ´image´       | Possible Image upload. The nodes ´backend_model´, ´upload_dir´ and ´base_url´ needs to be defined as well.        |

Reviewer: ```suggestion
| ´image´       | Possible image upload. The nodes ´backend_model´, ´upload_dir´ and ´base_url´ need to be defined as well.        |
```

Reviewer: 'password'
'file'
'label'
'allowspecific'

Need defining. Password is different to obscure.
